src/analysis/complex/phragmen_lindelof.lean

```diff
@@ -0,0 +1,579 @@
+      using hO.comp_tendsto (tendsto_neg_cobounded.inf H.tendsto) },
+  { rw [comp_app, ← of_real_neg],
+    exact hre (-x) (neg_nonneg.2 hx) },
+  { rw [comp_app, ← neg_mul, ← of_real_neg],
+    exact him (-x) (neg_nonpos.2 hx) }
+end
+
+/-- **Phragmen-Lindelöf principle** in the right half-plane. Let `f : ℂ → E` be a function such that
+
+* `f` is differentiable in the open right half-plane and is continuous on its closure;
+* `∥f z∥` is bounded from above by `A * exp(B * (abs z) ^ c)` on the open right half-plane
+  for some `c < 2`;
+* `∥f z∥` is bounded from above by a constant `C` on the imaginary axis;
+* `f x → 0` as `x : ℝ` tends to infinity.
+
+Then `∥f z∥` is bounded from above by the same constant on the closed right half-plane.
+See also `phragmen_lindelof.right_half_plane_of_bounded_on_real` for a stronger version. -/
+lemma right_half_plane_of_tendsto_zero_on_real (hd : diff_cont_on_cl ℂ f {z | 0 < z.re})
+  (hexp : ∃ (c < (2 : ℝ)) B, is_O f (λ z, expR (B * (abs z) ^ c))
+    (comap abs at_top ⊓ 𝓟 {z | 0 < z.re}))
+  (hre : tendsto (λ x : ℝ, f x) at_top (𝓝 0)) (him : ∀ x : ℝ, ∥f (x * I)∥ ≤ C) (hz : 0 ≤ z.re) :
+  ∥f z∥ ≤ C :=
+begin
+  revert z,
+  have hle : ∀ C', (∀ x : ℝ, 0 ≤ x → ∥f x∥ ≤ C') → ∀ z : ℂ, 0 ≤ z.re → ∥f z∥ ≤ max C C',
+  { intros C' hC' z hz,
+    cases le_total z.im 0,
+    { refine quadrant_IV (hd.mono $ λ _, and.left) (Exists₃.imp (λ c hc B hO, _) hexp)
+        (λ x hx, (hC' x hx).trans $ le_max_right _ _) (λ x hx, (him x).trans (le_max_left _ _))
+        ⟨hz, h⟩,
+      exact hO.mono (inf_le_inf_left _ $ principal_mono.2 $ λ _, and.left) },
+    { refine quadrant_I (hd.mono $ λ _, and.left) (Exists₃.imp (λ c hc B hO, _) hexp)
+        (λ x hx, (hC' x hx).trans $ le_max_right _ _) (λ x hx, (him x).trans (le_max_left _ _))
+        ⟨hz, h⟩,
+      exact hO.mono (inf_le_inf_left _ $ principal_mono.2 $ λ _, and.left) } },
+  obtain ⟨x, hx₀, hx⟩ : ∃ x : ℝ, 0 ≤ x ∧ ∀ y : ℝ, 0 ≤ y → ∥f y∥ ≤ ∥f x∥,
+  { have hfc : continuous_on (λ x : ℝ, f x) (Ici 0),
+    { refine hd.continuous_on.comp continuous_of_real.continuous_on (λ x hx, _),
+      rwa closure_set_of_lt_re },
+    by_cases h₀ : ∀ x : ℝ, 0 ≤ x → f x = 0,
+    { refine ⟨0, le_rfl, λ y hy, _⟩, rw [h₀ y hy, h₀ 0 le_rfl] },
+    push_neg at h₀,
+    rcases h₀ with ⟨x₀, hx₀, hne⟩,
+    have hlt : ∥(0 : E)∥ < ∥f x₀∥, by rwa [norm_zero, norm_pos_iff],
+    simpa only [exists_prop]
+      using hfc.norm.exists_forall_ge' is_closed_Ici hx₀ _,
+    rw [real.cocompact_eq, inf_sup_right, (disjoint_at_bot_principal_Ici (0 : ℝ)).eq_bot,
+      bot_sup_eq],
+    exact (hre.norm.eventually $ ge_mem_nhds hlt).filter_mono inf_le_left },
+  cases le_or_lt (∥f x∥) C,
+  { simpa only [max_eq_left h] using hle _ hx },
+  { have : is_max_on (norm ∘ f) {z | 0 < z.re} x,
+    { rintros z (hz : 0 < z.re),
+      simpa [max_eq_right h.le] using hle _ hx _ hz.le },
+    have : ∥f 0∥ = ∥f x∥,
+    { apply norm_eq_norm_of_is_max_on_of_closed_ball_subset hd this,
+      -- move to a lemma?
```

Comment:
```suggestion
```
I don't think it should be moved to a lemma, this is a pretty specific situation.